src/api/emailValidationApiClient.ts

```diff
@@ -0,0 +1,96 @@
+'use client';
+/* eslint-disable sonarjs/cognitive-complexity */
+import { isRight } from 'fp-ts/lib/Either';
+import { useCallback, useState } from 'react';
+import { createClient } from './generated/ioFunction/client';
+import { ValidateProfileEmailPayload } from './generated/ioFunction/ValidateProfileEmailPayload';
+import { ValidationToken } from './generated/ioFunction/ValidationToken';
+import { extractResponse, retryingFetch } from '@/app/[locale]/_utils/api-utils';
+
+const BASE_URL = `${process.env.NEXT_PUBLIC_IO_FUNCTION_API_BASE_URL}`;
+
+const validateEmailApiClient = createClient({
+  baseUrl: BASE_URL,
+  fetchApi: retryingFetch(),
+});
+
+const useFetch = () => {
```

Comment:
It it might be better to call it useFetchEmailValidation or a similar name to differentiate it from the existing useFetch hook